callosum/peer.py

```diff
@@ -256,13 +407,16 @@ def _lookup(self, msgtype, method):
                         rqst_id,
                         self._scheduler,
                         handler(request))
-                    result = await self._func_scheduler.wait(rqst_id)
+                    result = await self._func_scheduler.get_fut(rqst_id)
+                    self._func_scheduler.cleanup(rqst_id)
                 except (asyncio.TimeoutError, asyncio.CancelledError):
                     raise
                 except Exception as e:
                     self._log.error('Uncaught exception')
                     response = RPCMessage.error(request, e)
                 else:
+                    if 'cancelled' in result:
```

Comment:
If I understand your code correctly, `result` is an `asyncio.Future` object.
What does this mean?

Reply:
Yes, the result is an `asyncio.Future` object. In this case, my code is:
`result = await self._func_scheduler.get_fut(rqst_id)`
As it can be seen, I am awaiting on the future and thus getting the `result` of the future rather than `asyncio.Future` object.

I tested it and the cancellation is initiated. So it means that line `if 'cancelled' in result:` works as expected.

If you think that my code will be misleading to readers, just tell me and I will refactor it.

Reply:
So what is the type of `result`?
I'd like to express it in a more "type-safe" way, since it seems like that user-defined handlers may just include "cancelled" field for different purpose in the result, according to:
```python3
            if task.cancelled():
                result = {'cancelled': True}
                _resolve_future(rqst_id, fut, result, self._log)
            else:
                _resolve_future(rqst_id, fut, task.result(), self._log)
                heapq.heappop(self._pending[okey])
```

Reply:
Done